src/instana/instrumentation/kafka/confluent_kafka_python.py

```diff
@@ -105,78 +113,142 @@ def create_span(
         headers: Optional[List[Tuple[str, bytes]]] = [],
         exception: Optional[str] = None,
     ) -> None:
-        tracer, parent_span, _ = get_tracer_tuple()
-        parent_context = (
-            parent_span.get_span_context()
-            if parent_span
-            else tracer.extract(
-                Format.KAFKA_HEADERS,
-                headers,
-                disable_w3c_trace_context=True,
+        try:
+            span = consumer_span.get(None)
+            if span is not None:
+                close_consumer_span(span)
+
+            tracer, parent_span, _ = get_tracer_tuple()
+
+            if not tracer:
+                tracer = get_tracer()
+            is_suppressed = False
+
+            if topic:
+                is_suppressed = tracer.exporter._HostAgent__is_endpoint_ignored(
+                    "kafka",
+                    span_type,
+                    topic,
+                )
+
+            if not is_suppressed and headers:
+                for header_name, header_value in headers:
+                    if header_name == "x_instana_l_s" and header_value == b"0":
+                        is_suppressed = True
+                        break
+
+            if is_suppressed:
+                return
+
+            parent_context = (
+                parent_span.get_span_context()
+                if parent_span
+                else (
+                    tracer.extract(
+                        Format.KAFKA_HEADERS,
+                        headers,
+                        disable_w3c_trace_context=True,
+                    )
+                    if tracer.exporter.options.kafka_trace_correlation
+                    else None
+                )
+            )
+            span = tracer.start_span(
+                "kafka-consumer", span_context=parent_context, kind=SpanKind.CONSUMER
             )
-        )
-        with tracer.start_as_current_span(
-            "kafka-consumer", span_context=parent_context, kind=SpanKind.CONSUMER
-        ) as span:
             if topic:
                 span.set_attribute("kafka.service", topic)
             span.set_attribute("kafka.access", span_type)
-
             if exception:
                 span.record_exception(exception)
+                span.end()
+
+            save_consumer_span_into_context(span)
+        except Exception as e:
+            logger.debug(f"Error while creating kafka-consumer span: {e}")
+
+    def save_consumer_span_into_context(span: "InstanaSpan") -> None:
+        ctx = trace.set_span_in_context(span)
+        token = context.attach(ctx)
+        consumer_token["token"] = token
```

Comment:
If changing the `consumer_token` variable as commented [here](https://github.com/instana/python-sensor/pull/790/files#r2302190833), you can do the following:
```suggestion
        consumer_token = context.attach(ctx)
```